From 9344e5bf9fac766de4362943d074875216c8ae12 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 5 May 2015 09:55:30 +0800
Subject: [PATCH] MLK-10813-2: ASoC: imx-wm8958: fix issue that clock is not
 disabled

The clock disable has no chance to be called in set_bias_level_post,
because the dapm->bias_level is set by the codec driver, machine driver
should use the card->dapm.bias_level.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 sound/soc/fsl/imx-wm8958.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index 3675d0c..9be2362 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -353,7 +353,7 @@ static int imx_wm8958_set_bias_level(struct snd_soc_card *card,
 
 	switch (level) {
 	case SND_SOC_BIAS_STANDBY:
-		if (dapm->bias_level == SND_SOC_BIAS_OFF) {
+		if (card->dapm.bias_level == SND_SOC_BIAS_OFF) {
 			if (!IS_ERR(data->mclk)) {
 				ret = clk_prepare_enable(data->mclk);
 				if (ret) {
@@ -382,7 +382,7 @@ static int imx_wm8958_set_bias_level_post(struct snd_soc_card *card,
 
 	switch (level) {
 	case SND_SOC_BIAS_OFF:
-		if (dapm->bias_level == SND_SOC_BIAS_STANDBY)
+		if (card->dapm.bias_level == SND_SOC_BIAS_STANDBY)
 			if (!IS_ERR(data->mclk))
 				clk_disable_unprepare(data->mclk);
 		break;
@@ -390,7 +390,7 @@ static int imx_wm8958_set_bias_level_post(struct snd_soc_card *card,
 		break;
 	}
 
-	dapm->bias_level = level;
+	card->dapm.bias_level = level;
 
 	return 0;
 }
-- 
1.8.0

